src/services/db/CommitServiceGitFile.ts

```diff
@@ -0,0 +1,84 @@
+import UserWithSiteSessionData from "@root/classes/UserWithSiteSessionData"
+import isFileAsset from "@root/utils/commit-utils"
+import { isReduceBuildTimesWhitelistedRepo } from "@root/utils/growthbook-utils"
+
+import GitFileSystemService from "./GitFileSystemService"
+import { STAGING_BRANCH } from "./GitHubService"
+
+/**
+ * Responsibilities of this class
+ * 1. Creates all commits to staging
+ * 2. Creates non-asset related commits to staging-lite
+ */
+export default class CommitServiceGitFile {
+  private readonly STAGING_LITE_BRANCH = "staging-lite"
+
+  private readonly gitFileSystemService: GitFileSystemService
+
+  constructor(gitFileSystemService: GitFileSystemService) {
+    this.gitFileSystemService = gitFileSystemService
+  }
+
+  async create(
+    sessionData: UserWithSiteSessionData,
+    {
+      content,
+      fileName,
+      directoryName,
+      isMedia = false,
+    }: {
+      content: string
+      fileName: string
+      directoryName: string
+      isMedia?: boolean
+    }
+  ): Promise<{ sha: string }> {
```

Comment:
Are we able to use `neverthrow` here?

Reply:
yes but not now pls post release can refactor